src/dda/dda.c

```diff
@@ -0,0 +1,87 @@
+/* ************************************************************************** */
+/*                                                                            */
+/*                                                        :::      ::::::::   */
+/*   dda.c                                              :+:      :+:    :+:   */
+/*                                                    +:+ +:+         +:+     */
+/*   By: ycarro <ycarro@student.42.fr>              +#+  +:+       +#+        */
+/*                                                +#+#+#+#+#+   +#+           */
+/*   Created: 2023/02/22 14:48:17 by ycarro            #+#    #+#             */
+/*   Updated: 2023/02/23 17:42:22 by ycarro           ###   ########.fr       */
+/*                                                                            */
+/* ************************************************************************** */
+
+#include "dda.h"
+
+void	dda(char **map, t_player *p, t_ray	*ray_data, t_point *collision)
+{
+	(void)map;
+	(void)collision;
+	set_incrementor(ray_data, ray_data->to_cast.radians);
+	get_ray_data(ray_data, ray_data->to_cast.radians, p);
+	while (!it_collides(ray_data, map, collision))
+	{
+		if (ray_differs_in_x(ray_data))
+		{
+			ray_data->x_pos.x += (double)(ray_data->incrementor.x * 100);
+			ray_data->x_pos.y = (ray_data->slope * ray_data->x_pos.x \
+				+ ray_data->intercept);
+		}
+		else
+		{
+			ray_data->y_pos.x += (double)(ray_data->incrementor.x * 100);
+			ray_data->y_pos.y = (ray_data->slope * ray_data->x_pos.x \
+				+ ray_data->intercept);
+		}
+	}
+}
+
+void	set_incrementor(t_ray *ray_data, double alpha)
+{
+	while (alpha > (2 * M_PI))
+		alpha -= (2 * M_PI);
+	if (alpha >= (3.0 / 2.0) * M_PI)
+	{
+		ray_data->incrementor.x = 1;
+		ray_data->incrementor.y = -1;
+	}
+	else if (alpha >= M_PI)
+	{
+		ray_data->incrementor.x = -1;
+		ray_data->incrementor.y = -1;
+	}
+	else if (alpha >= (1.0 / 2.0) * M_PI)
+	{
+		ray_data->incrementor.x = -1;
+		ray_data->incrementor.y = 1;
+	}
+	else
+	{
+		ray_data->incrementor.x = 1;
+		ray_data->incrementor.y = 1;
+	}
+}
+
+void	get_ray_data(t_ray *ray_data, double alpha, t_player *player)
+{
+	ray_data->slope = tan(alpha);
+	ray_data->intercept = player->pos.y - (player->pos.x * ray_data->slope);
+	ray_data->x_pos.x = player->pos.x;
+	ray_data->x_pos.y = player->pos.y;
+	ray_data->y_pos.x = player->pos.x;
+	ray_data->y_pos.x = player->pos.y;
+}
+
+int	ray_differs_in_x(t_ray *ray_data)
+{
+	if (ray_data->incrementor.x == 1)
+	{
+		if ((int)ray_data->x_pos.x <= (int)ray_data->y_pos.x)
+			return (1);
+	}
+	if (ray_data->incrementor.x == -1)
+	{
+		if ((int)ray_data->x_pos.x >= (int)ray_data->y_pos.x)
+			return (1);
+	}
+	return (0);
```

Comment:
when do we return 0?
can ray_data->incrementor.x be something else then 1 & -1?

Reply:
Incrementor only sets the direction we are moving in the x and y axis, so 1 is moving forward and -1 moving backward. This will depend on the angle we are looking. If we are looking to the north then incrementor.x wil be 1 and incrementor.y will be -1
So, no, it only will have 1 and -1 values. I set the like that because I need the positive or negative value for later the ray_increment part
0 is returned when y ray is bigger than x ray in the x axis, so ray_increment knows wich of the rays increment